sycl/test/kernel-and-program/cache.cpp

```diff
@@ -243,15 +253,16 @@ static void testKernelCachePositive() {
     auto *CLKernel = cl::sycl::detail::getSyclObjImpl(Kernel)->getHandleRef();
 
     auto *Ctx = cl::sycl::detail::getRawSyclObjImpl(Prog.get_context());
+    const auto &Cache = Ctx->getKernelProgramCache();
 
-    assert(Ctx->getCachedKernels().size() == 1 &&
+    assert(Cache.getKernelsPerProgramCache().size() == 1 &&
            "Expecting only a single element in kernels cache");
-    assert(Ctx->getCachedKernels().begin()->first ==
+    assert(Cache.getKernelsPerProgramCache().begin()->first ==
                pi::cast<pi_program>(CLProg) &&
            "Invalid program key in kernels cache");
-    assert(Ctx->getCachedKernels().begin()->second.size() == 1 &&
+    assert(Cache.getKernelsPerProgramCache().begin()->second.size() == 1 &&
            "Expecting only a single kernel for the program");
-    assert(Ctx->getCachedKernels().begin()->second.begin()->second ==
+    assert(Cache.getKernelsPerProgramCache().begin()->second.begin()->second.Ptr.load() ==
```

Comment:
This is terrible.

Reply:
What do you mean?

Reply:
I don't understand what this line does. Please use temporary variables to hold values of `second` objects to make this more readable.